src/mqtt.c

```diff
@@ -238,6 +238,20 @@ static void on_message(
   sfree(vl.values);
 } /* void on_message */
 
+static int mqtt_subscribe(mqtt_client_conf_t *conf) {
```

Comment:
I'm not convinced this needs its own function: the function is ~13 lines long and saves ~4 lines. I think the code would be shorter and easier to read without it.

Reply:
Well, I think it's also about the code readability and maintainability.
Calling `mqtt_subscribe(conf)` two times looks nicer to me than calling `mosquitto_subscribe(conf->mosq, /* message_id = */ NULL, conf->topic, conf->qos)` two times.
And duplicating the code would duplicate not only the `mosquitto_subscribe() `call but also the whole error message.

Reply:
Agreed, it looks nicer. I'd argue the (less nice) code using `mosquitto_subscribe()` directly would be more readable, because all relevant information is immediately visible. Using the wrapper obfuscates this information.

I don't want to spend too much time on something with little (if any) gain, so let's go with the function.